.gitlab-ci.jsonnet

```diff
@@ -1,202 +0,0 @@
-            docker.rename(images.e2e.name, images.e2elatest.name),
-        only: ['tags'],
-    },
-
-    'wait-in-queue': baseJob.WaitInQueue {
-        stage: stages.wait_in_queue,
-    },
-    
-    'e2e-setup': baseJob.Deploy {
-        local _vars = self.localvars,
-        localvars+:: {
-            namespace: "e2e-%s" % "${CI_COMMIT_REF_SLUG}-${SHA8}",
-            catalog_namespace: "e2e-%s" % "${CI_COMMIT_REF_SLUG}-${SHA8}",
-        },
-        stage: stages.test_setup,
-    },
-
-    'e2e-teardown': baseJob.DeployStop {
-        local _vars = self.localvars,
-        localvars+:: {
-            namespace: "e2e-%s" % "${CI_COMMIT_REF_SLUG}-${SHA8}",
-            catalog_namespace: "e2e-%s" % "${CI_COMMIT_REF_SLUG}-${SHA8}",
-        },
-        stage: stages.test_teardown,
-    },
-
-    // End2End tests
-    local integration_test = baseJob.EndToEndTest {
-        stage: stages.tests,
-    },
-
-    e2e_tests: integration_test {
-    },
-
-    "deploy-preview": baseJob.Deploy {
-        local _vars = self.localvars,
-        localvars+:: {
-            helm_opts: ["--force"],
-        },
-        stage: stages.deploy_preview,
-        when: "manual",
-        environment+: {
-            on_stop: "stop-preview",
-        },
-    } + onlyBranch,
-
-    "stop-preview": baseJob.DeployStop {
-        when: "manual",
-        stage: stages.deploy_preview,
-    } + onlyBranch,
-
-    "deploy-staging": baseJob.Deploy {
-        local _vars = self.localvars,
-        localvars+:: {
-            image: images.release,
-            domain: "alm-staging.k8s.devtable.com",
-            namespace: "ci-alm-staging",
-            channel: "staging",
-            helm_opts: ["--force"],
-            kubeconfig: "$CD_KUBECONFIG",
-        },
-        stage: stages.deploy_staging,
-        script+: [],
-        environment+: {
-            name: "staging",
-        },
-        only: ['master'],
-    },
-
-    # currently not needed, to re-enable remove the "." prefix
-    ".deploy-teamui": baseJob.Deploy {
-        local _vars = self.localvars,
-        localvars+:: {
-            image: images.release,
-            domain: "teamui.console.team.coreos.systems",
-            namespace: "operator-lifecycle-manager",
-            catalog_namespace: "operator-lifecycle-manager",
-            channel: "staging",
-            helm_opts: ["--force"],
-            kubeconfig: "$TEAMUI_KUBECONFIG",
-            params+:: {
-                watchedNamespaces: "",
-            },
-        },
-        stage: stages.deploy_staging,
-        script+: [
-            "curl -X POST --data-urlencode \"payload={\\\"text\\\": \\\"New OLM Operator quay.io/coreos/olm:${CI_COMMIT_REF_SLUG}-${CI_COMMIT_SHA} deployed to ${TEAMUI_HOST}/k8s/ns/operator-lifecycle-manager/deployments/alm-operator\\\"}\" ${TEAMUI_SLACK_URL}",
-        ],
-        environment+: {
-            name: "teamui",
-        },
-        only: [''],
-    },
-
-    "deploy-openshift": baseJob.Deploy {
```

Comment:
How can we do this using Prow?

Reply:
When the e2e tests run, you can login to the cluster it spins up if that's good enough.

If you want a cluster that won't disappear after the tests, you can pull the `origin-release` image built by the PR and start a cluster with installer (locally with libvirt or remotely in AWS)

It's not _as_ easy, but it's not particularly difficult either. If we miss this too much we can invest some time making additional tide commands to do this for us maybe?

Reply:
We need `master` of OLM to be deployed to the same persistent cluster where the end-to-end tests for `openshift/console` runs.

Reply:
We can keep it, but console just needs to run e2e tests in openshift ci and it will automatically get tested against master. That seems easier to maintain long term

Reply:
@alecmerdler I'm wondering why the `openshift/console` ci doesn't deploy the `master` version of OLM to its persistent cluster. It seems like the responsibility should be inverted from what we had before.

Reply:
> console just needs to run e2e tests in openshift ci and it will automatically get tested against master

True, but we can't really talk since this literal PR is what takes us off our own ad-hoc CI.

> I'm wondering why the openshift/console ci doesn't deploy the master version of OLM to its persistent cluster

Keyword here is `persistent`. That 3.11 cluster is spun up every once in a while (when it eventually dies) so doesn't receive updates.